File: spec/requests/static_spec.rb
# this is the Ruby counterpart to spec/javascripts/static_spec.js
# here we're testing the /static controller works

describe "/static/*" do
  it "returns test template" do
    get "/static/test"
    expect(response.status).to eq(200)
    expect(response.body).to start_with('<!--')
  end

  it "renders haml template" do
    get "/static/test_haml"
    expect(response.status).to eq(200)
    expect(response.body).to match(/<div class=['"]testclass['"]>/)
    expect(response.body).not_to include(".testclass")
  end
end

File: spec/javascripts/static_spec.js
// this is the JS counterpart to spec/requests/static_spec.rb
// here we're testing if the jasmine integration works

describe('/static/* in jasmine', function() {
  it("returns test template", function(done) {
    Promise.resolve($.get("/static/test.html"))
      .then(function(data) {
        expect(data).toMatch(/^<!--/);
      })
      .catch(function() {
        expect("Error").toEqual(null);
      })
      .then(done);
  });

  it("renders haml template", function(done) {
    Promise.resolve($.get("/static/test_haml.html.haml"))
      .then(function(data) {
        expect(data).toMatch(/<div class=['"]testclass['"]>/);
        expect(data).not.toMatch(/\.testclass/);
      })
      .catch(function() {
        expect("Error").toEqual(null);
      })
      .then(done);
  });
});
